src/Uno.UI.Runtime.Skia.Gtk/Input/GtkCorePointerInputSource.cs

```diff
@@ -195,6 +195,19 @@ private void OnLeaveEvent(object o, LeaveNotifyEventArgs args)
 
 	private void OnButtonPressEvent(object o, ButtonPressEventArgs args)
 	{
+		// Double clicking on Gtk will produce an event sequence with the following event types on Gtk:
+		// 1. EventType.ButtonPress
+		// 2. EventType.ButtonRelease
+		// 3. EventType.ButtonPress
+		// 4. EventType.TwoButtonPress
+		// 5. EventType.ButtonRelease
+		// We want to receive the "single" presses only as receiving the second press twice is gonna be problematic.
+		// So we skip TwoButtonPress (and also ThreeButtonPress as it has similar issue)
+		if (args.Event.Type is EventType.TwoButtonPress or EventType.ThreeButtonPress)
```

Comment:
To avoid events like FourButtonPress / FiveButtonPress that could be added in a future version of GTK should we just allow type that are known?

```suggestion
		if (args.Event.Type is not EventType.ButtonPress and not EventType.ButtonRelease)
```

Reply:
There is `PadButtonPress` that could probably be excluded incorrectly. We could do it as `args.Event.Type is not (EventType.ButtonPress or EventType.PadButtonPress)` but I'm not really sure how good that will be, or if it's arguable that other kind of presses that we *do care* about might be added in future as well.

Will leave the final call to you on which path to take.

Reply:
I'm good with the current approach, especially if there are some other exotic value like `PadButtonPress`.